Let staticcheck take args

valentin-krasontovitsch opened the pull request:
this MR hopes to resolve #11 by allowing staticcheck to take command line arguments via pre-commit's `args` option.

see commit messages for further details.

valentin-krasontovitsch committed:
feat: accept args for staticcheck

staticcheck needs some massaging of the input files given to it by
pre-commit - while staticcheck can run on single files, it will complain
if there are missing dependencies that get resolved in another file in
the same package. Furthermore, if it is passed several files, it demands
that those files live in the same package.

hence the approach has been to parse the file arguments and transform
them into their respective packages, calling staticcheck on one package
at a time.

the simple original solution did not allow for arguments to staticcheck
to be passed via pre-commit's `args` option.

we iterate on the original solution by parsing the passed arguments,
separating possible command line options and file arguments.

resolves #11

valentin-krasontovitsch committed:
feat: run staticcheck in serial mode

the way we collate all file arguments into their respective packages,
parallelization by pre-commit is counter-productive, as files from the
same packages can end up in different invocations of staticcheck,
resulting in the same package being run several times.

Bahjat commented:
@valentin-krasontovitsch thanks a lot for opening this PR.
Could you please rebase with the master and run the shellcheck on your added shell file?

valentin-krasontovitsch commented:
sure thing! thank you for responding : )

i funnily in the course of working on this script setup shellcheck and integrated it with my editor ^^

rebased, fixed up a correction commit, and pushed just now.

apparently the workflow needs approval before it will be run...
